test/nextjs.test.js

```diff
@@ -606,6 +607,217 @@ describe(`NextJS Middleware Test: ${printPath("[test/nextjs.test.js]")}`, functi
+        assert.equal(sessionContainer.hasToken, false);
+        assert.equal(sessionContainer.session, undefined);
+
+        const authenticatedRequest = new NextRequest("http://localhost:3000/api/get-user", {
+            headers: {
+                Authorization: `Bearer ${tokens.access}`,
+            },
+        });
+
+        sessionContainer = await getSSRSession(authenticatedRequest.cookies.getAll(), authenticatedRequest.headers);
+
+        assert.equal(sessionContainer.hasToken, true);
+        assert.equal(sessionContainer.session.getUserId(), process.env.user);
+    });
+
+    it("withSession", async function () {
+        const handleCall = getAppDirRequestHandler(NextResponse);
+
+        const userEmail = `john.doe.${Date.now()}@supertokens.com`;
+        const requestInfo = {
+            method: "POST",
+            headers: { rid: "emailpassword" },
+            body: JSON.stringify({
+                formFields: [
+                    { id: "email", value: userEmail },
+                    { id: "password", value: "P@sSW0rd" },
+                ],
+            }),
+        };
+
+        const signUpRequest = new NextRequest("http://localhost:3000/api/auth/signup", requestInfo);
+
+        const signUpRes = await handleCall(signUpRequest);
+        assert.deepStrictEqual(signUpRes.status, 200);
+        const tokens = getSessionTokensFromResponse(signUpRes);
+        assert.notEqual(tokens.access, undefined);
+        assert.notEqual(tokens.refresh, undefined);
+
+        const unAuthenticatedRequest = new NextRequest("http://localhost:3000/api/get-user");
+
+        const unAuthenticatedResponse = await withSession(unAuthenticatedRequest, async (session) => {
+            if (!session) {
+                return new NextResponse("Authentication required", { status: 401 });
+            }
+
+            return NextResponse.json({
+                userId: session.getUserId(),
+                sessionHandle: session.getHandle(),
+                accessTokenPayload: session.getAccessTokenPayload(),
+            });
+        });
+
+        assert.equal(unAuthenticatedResponse.status, 401);
+
+        const authenticatedRequest = new NextRequest("http://localhost:3000/api/get-user", {
+            headers: {
+                Authorization: `Bearer ${tokens.access}`,
+            },
+        });
+
+        const authenticatedResponse = await withSession(authenticatedRequest, async (session) => {
+            if (!session) {
+                return new NextResponse("Authentication required", { status: 401 });
+            }
+
+            return NextResponse.json({
+                userId: session.getUserId(),
+                sessionHandle: session.getHandle(),
+                accessTokenPayload: session.getAccessTokenPayload(),
+            });
+        });
+
+        assert.equal(authenticatedResponse.status, 200);
+        const json = await authenticatedResponse.json();
+        assert(json.userId === process.env.user);
+    });
```

Comment:
also add a test in which you pass in an expired access token and check that that returns a 401 with try refresh token error

Reply:
And also one for invalid claims - it should return a 403

Reply:
done